src/Raven.Server/Indexing/TempFileCache.cs

```diff
@@ -214,8 +215,27 @@ public override void SetLength(long value)
         public override void Write(byte[] buffer, int offset, int count)
         {
             var pos = InnerStream.Position;
-            InnerStream.Write(buffer, offset, count);
-            _length = Math.Max(_length, pos + count);
+            try
+            {
+                InnerStream.Write(buffer, offset, count);
+                _length = Math.Max(_length, pos + count);
+            }
+            catch (IOException e)
+            {
+                if (IsDiskFull(e))
```

Comment:
See `IsOutOfDiskSpaceException` in `ExceptionHelper`.

Reply:
Done